etl/steps/data/garden/lgbt_rights/2024-06-03/equaldex.py

```diff
@@ -0,0 +1,244 @@
+        "No censorship": "No censorship",
+        "Varies by Region": "Varies by region",
+        "Ambiguous": "Ambiguous",
+        "Other punishment": "Other punishment",
+        "Fine as punishment": "Fine as punishment",
+        "State-enforced": "State-enforced",
+        "Imprisonment as punishment": "Imprisonment as punishment",
+    },
+    "conversion_therapy": {
+        "Banned": "Banned",
+        "Varies by Region": "Varies by region",
+        "Sexual orientation only": "Sexual orientation only",
+        "Ambiguous": "Ambiguous",
+        "Not banned": "Not banned",
+    },
+    "discrimination": {
+        "Illegal": "Illegal",
+        "Illegal in some contexts": "Illegal in some contexts",
+        "Varies by Region": "Varies by region",
+        "Ambiguous": "Ambiguous",
+        "No protections": "No protections",
+    },
+    "employment_discrimination": {
+        "Sexual orientation and gender identity": "Sexual orientation and gender identity",
+        "Gender identity only": "Gender identity only",
+        "Sexual orientation only": "Sexual orientation only",
+        "Varies by Region": "Varies by region",
+        "Ambiguous": "Ambiguous",
+        "No protections": "No protections",
+    },
+    "housing_discrimination": {
+        "Sexual orientation and gender identity": "Sexual orientation and gender identity",
+        "Gender identity only": "Gender identity only",
+        "Sexual orientation only": "Sexual orientation only",
+        "Varies by Region": "Varies by region",
+        "Ambiguous": "Ambiguous",
+        "No protections": "No protections",
+    },
+    "military": {
+        "Legal": "Legal",
+        "Don't Ask, Don't Tell": "\"Don't ask, don't tell\"",
+        "Lesbians, gays, bisexuals permitted, transgender people banned": "LGB permitted, transgender people banned",
+        "Ambiguous": "Ambiguous",
+        "Illegal": "Illegal",
+    },
+    "non_binary_gender_recognition": {
+        "Recognized": "Recognized",
+        "Intersex only": "Intersex only",
+        "Varies by Region": "Varies by region",
+        "Ambiguous": "Ambiguous",
+        "Not legally recognized": "Not legally recognized",
+    },
+    "gender_affirming_care": {
+        "Legal": "Legal",
+        "Legal, but restricted for minors": "Legal, but restricted for minors",
+        "Legal, but banned for minors": "Legal, but banned for minors",
+        "Varies by Region": "Varies by region",
+        "Ambiguous": "Ambiguous",
+        "Restricted": "Restricted",
+        "Banned": "Banned",
+    },
+    "intersex_infant_surgery": {
+        "Not banned": "Not banned",
+        "Parental approval required": "Parental approval required",
+        "Varies by Region": "Varies by region",
+        "Ambiguous": "Ambiguous",
+        "Full ban": "Full ban",
+    },
+}
+
+
+def process_data(tb: Table) -> Table:
+    """Process data from Equaldex."""
+
+    # Make the dataframe wide by pivoting on the issue column.
```

Comment:
@Marigold When I add this 
tb["value_formatted"] = tb["value_formatted"].astype(str)

I have the problem with the sort again (this one https://github.com/owid/etl/pull/2644)

Is there a way to fix it? Thanks!

Reply:
The problem is that there are NaN values in `value_formatted` that get converted to string `nan` when you call `astype(str)`. You need to keep them as NaN. There are two ways how to do it:
```
tb["value_formatted"] = tb["value_formatted"].astype("string")
```
or
```
tb["value_formatted"] = tb["value_formatted"].astype(str).mask(tb["value_formatted"].isnull(), np.nan)
```

(I haven't been using `"string"` much, so it's possible it might not work)

Reply:
Thank you!